src/proof/proof_node.h

```diff
@@ -133,6 +134,26 @@ inline size_t ProofNodeHashFunction::operator()(
   return pfn->getResult().getId() + static_cast<unsigned>(pfn->getRule());
 }
 
+inline size_t ProofNodeHashFunction::operator()(const ProofNode* pfn) const
+{
+  uint64_t ret = fnv1a::offsetBasis;
+
+  ret = fnv1a::fnv1a_64(ret, std::hash<Node>()(pfn->getResult()));
+  ret = fnv1a::fnv1a_64(ret, static_cast<size_t>(pfn->getRule()));
+
+  for (const cvc5::Pf& child : pfn->getChildren())
```

Comment:
Can't it be 
```suggestion
  for (const Pf& child : pfn->getChildren())
```
?

Also store `pfn->getChildren()` into a constant reference that you use in the loop. The same for `pfn->getArguments()` below.